tests/unittest/llmapi/apps/_test_openai_responses_perf_metrics.py

```
@@ -0,0 +1,357 @@
+    with RemoteOpenAIServer(model_path, args) as remote_server:
+        yield remote_server
+        logger.info("Tests completed, shutting down server")
+
+
+# ---------------------------------------------------------------------------
+# Helpers
+# ---------------------------------------------------------------------------
+
+def _fetch_perf_metrics(server: RemoteOpenAIServer) -> list:
+    """Return the current /perf_metrics list (may be empty)."""
+    response = urlopen(f"{server.url_root}/perf_metrics")
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)